source/scripts/Background/controllers/account/evm.ts

```diff
@@ -71,8 +81,67 @@ const EthAccountController = (): IEthAccountController => {
     }
   };
 
+  const editTokenInfo = (token: ITokenEthProps) => {
+    try {
+      const { activeAccount, accounts } = store.getState().vault;
+
+      const cloneArray = cloneDeep(
+        accounts[activeAccount.type][activeAccount.id].assets
+      );
+
+      const findIndex = cloneArray.ethereum.findIndex(
+        (stateToken) => stateToken.contractAddress === token.contractAddress
+      );
+
+      store.dispatch(
+        setEditedEvmToken({
+          accountType: activeAccount.type,
+          accountId: activeAccount.id,
+          tokenIndex: findIndex,
+          editedToken: token,
+        })
+      );
+    } catch (error) {
+      throw new Error(`Could not edit token info. Error: ${error}`);
+    }
+  };
+
+  const deleteTokenInfo = (tokenAddress: string) => {
+    try {
+      const { activeAccount, accounts } = store.getState().vault;
+
+      const tokenExists = accounts[activeAccount.type][
+        activeAccount.id
+      ].assets.ethereum?.find(
+        (asset: ITokenEthProps) => asset.contractAddress === tokenAddress
+      );
+
+      if (!tokenExists) throw new Error("Token doesn't exists!");
+
+      store.dispatch(
+        setAccountPropertyByIdAndType({
+          id: activeAccount.id,
+          type: activeAccount.type,
+          property: 'assets',
+          value: {
+            ...accounts[activeAccount.type][activeAccount.id].assets,
+            ethereum: accounts[activeAccount.type][
```

Comment:
move this into a const, it will make it easier to read. Also, you may need to clone deep before saving the array on the state again.